src/js/libgeo/GeoOps.js

```diff
@@ -946,27 +946,27 @@ geoOps.SelectConic.updatePosition = function(el) {
 
 // conic by 4 Points and 1 line
 geoOps._helper.ConicBy4p1l = function(el, a, b, c, d, l) {
+    var al = List.scalproduct(a, l);
+    var bl = List.scalproduct(b, l);
+    var cl = List.scalproduct(c, l);
+    var dl = List.scalproduct(d, l);
+    var bcd = List.det3(b, c, d);
+    var abd = List.det3(a, b, d);
+    var acd = List.det3(a, c, d);
+    var abc = List.det3(a, b, c);
+    var mul = CSNumber.mult;
+    var r1 = CSNumber.sqrt(mul(mul(bl, dl), mul(bcd, abd)));
+    var r2 = CSNumber.sqrt(mul(mul(al, cl), mul(acd, abc)));
     var a1 = List.cross(List.cross(a, c), l);
     var a2 = List.cross(List.cross(b, d), l);
-    var b1 = List.cross(List.cross(a, b), l);
-    var b2 = List.cross(List.cross(c, d), l);
-    var o = List.realVector(csport.to(100 * Math.random(), 100 * Math.random()));
-
-    var r1 = CSNumber.mult(List.det3(o, a2, b1), List.det3(o, a2, b2));
-    r1 = CSNumber.sqrt(r1);
-    var r2 = CSNumber.mult(List.det3(o, a1, b1), List.det3(o, a1, b2));
-    r2 = CSNumber.sqrt(r2);
-
     var k1 = List.scalmult(r1, a1);
     var k2 = List.scalmult(r2, a2);
-
-    var x = List.add(k1, k2);
-    var y = List.sub(k1, k2);
-
-    var t1 = geoOps._helper.ConicBy5(el, a, b, c, d, x);
-    var t2 = geoOps._helper.ConicBy5(el, a, b, c, d, y);
-
-    return [t1, t2];
+    var x = List.normalizeMax(List.add(k1, k2));
+    var y = List.normalizeMax(List.sub(k1, k2));
+    var xy = tracing2(x, y);
+    var t1 = geoOps._helper.ConicBy5(el, a, b, c, d, xy.value[0]);
+    var t2 = geoOps._helper.ConicBy5(el, a, b, c, d, xy.value[1]);
+    return [List.normalizeMax(t1), List.normalizeMax(t2)];
```

Comment:
`List.scalproduct`, `List.cross`, `List.det3` and `List.normalizeMax` are used four times each. References to these `List` functions could all be localized like so:
```javascript
	var sp = List.scalproduct;
	var cp = List.cross;
	var det3 = List.det3;
	var nm = List.normalizeMax;
    var al = sp(a, l);
    var bl = sp(b, l);
    var cl = sp(c, l);
    var dl = sp(d, l);
    var bcd = det3(b, c, d);
    var abd = det3(a, b, d);
    var acd = det3(a, c, d);
    var abc = det3(a, b, c);
    var mul = CSNumber.mult;
    var r1 = CSNumber.sqrt(mul(mul(bl, dl), mul(bcd, abd)));
    var r2 = CSNumber.sqrt(mul(mul(al, cl), mul(acd, abc)));
    var a1 = cp(cp(a, c), l);
    var a2 = cp(cp(b, d), l);
    var k1 = List.scalmult(r1, a1);
    var k2 = List.scalmult(r2, a2);
    var x = nm(List.add(k1, k2));
    var y = nm(List.sub(k1, k2));
    var xy = tracing2(x, y);
    var t1 = geoOps._helper.ConicBy5(el, a, b, c, d, xy.value[0]);
    var t2 = geoOps._helper.ConicBy5(el, a, b, c, d, xy.value[1]);
    return [nm(t1), nm(t2)];
```